packages/marketplace/src/components/ui/developer-analytics/billing/transaction-history/transaction-history.tsx

```diff
@@ -1,44 +1,144 @@
 import * as React from 'react'
-import { H4, Grid, GridItem, H6 } from '@reapit/elements'
+import { useSelector } from 'react-redux'
+import { H4, Grid, GridItem, H6, fetcher, Loader } from '@reapit/elements'
 import styles from '@/styles/pages/developer-analytics.scss?mod'
+import { selectDeveloperApps } from '@/selector/developer'
+import { ReduxState } from '@/types/core'
+import lodashIsEqual from 'lodash.isequal'
+import { formatRenderDate, formatRequestDate, generatePreviousTransactionDate } from './utils'
+import dayjs, { Dayjs } from 'dayjs'
+import { URLS, generateHeader } from '@/constants/api'
 
 export type TransactionHistoryProps = {}
 
+export type MapState = {
+  isLoadingDeveloperDetail: boolean
+  developerCreatedDate: string
+  developerAppIds: string[]
+}
+
+export const selectTransactionHistoryState: (state: ReduxState) => MapState = state => {
+  const developerApps = selectDeveloperApps(state)
+
+  const developerAppIds = developerApps
+    .map(developerApp => developerApp.id)
+    .filter(id => typeof id === 'string') as string[]
+
+  return {
+    developerCreatedDate: state.developerDetail?.developerDetailData?.created || '',
+    developerAppIds: developerAppIds,
+    isLoadingDeveloperDetail: state.developerDetail.loading,
+  }
+}
+
+export const createHandleDownLoadButtonOnClickFn = ({
+  month,
+  developerAppIds,
+}: {
+  month: string
+  developerAppIds: string[]
+}) => async e => {
+  e.preventDefault()
+
+  const param = new URLSearchParams()
+
+  param.append('applicationId', developerAppIds.join(','))
+  const res = (await fetcher({
```

Comment:
pls use create new function named `downloader` for download the item instead of using fetcher